File: scala/anagram/src/main/scala/Anagram.scala
class Anagram(word: String) {
  def matches(candidates: Seq[String]): Seq[String] =
    candidates filter (isAnagram(word, _))

  private def isAnagram(word: String, candidate: String): Boolean = {
    val w = word.toLowerCase
    val c = candidate.toLowerCase
    w != c && w.sorted == c.sorted
  }
}

File: erlang/anagram/anagram.erl
-module(anagram).
-export([find/2]).

find(Word, Candidates) ->
  lists:filter(fun(C) -> is_anagram(Word, C) end, Candidates).

is_anagram(Word, Candidate) ->
  W = string:to_lower(Word),
  C = string:to_lower(Candidate),
  W =/= C andalso lists:sort(W) =:= lists:sort(C).
